SOOP/SOOP_CO2/create_CO2_netcdf_from_txt.py

```diff
@@ -0,0 +1,344 @@
+    # WSP converted to m s-1
+    WSPD_raw[:] = dataf['MetTrueWindSpKts'].values
+    WDIR_raw[:] = dataf['MetTrueWindDir'].values
+
+    ncfile.close()
+
+def read_realtime_file(rt_file):
+    """
+    Reads in data from realtime text files
+    Cause input files have inconsistent number of column per line
+    the function first reads file line by line into list, then create data frame
+    Returns  : dataframe
+               vessel_code_short
+    """
+    data = []
+
+    file_basename = os.path.basename(rt_file)
+
+    # check that vessel specific prefix is valid
+    platform_code_short = file_basename[0:2]
+    assert platform_code_short in VESSEL, "File name '%s' has unknown vessel_code" % rt_file
+    platform_code = VESSEL[platform_code_short]
+
+    with open(rt_file, 'r') as f:
+        header_txt_file = f.readline()
+        header_txt_file = (re.split(r'\t+', header_txt_file.rstrip('\r\n')))
+
+        # create  an ordered dictionary of paramter name:indices
+        headers = [(header_col, header_txt_file.index(header_col))
+                for header_col in header_txt_file]
+        input_rt_parameter = collections.OrderedDict(headers)
+
+        for line in f:
+            tmp_line = re.split(r'\t+', line.rstrip('\r\n'))
+            if len(tmp_line) == len(input_rt_parameter):
+                data.append(tmp_line)
+            elif len(tmp_line) >5 and len(tmp_line) <len(input_rt_parameter):
+                line_filled_w_nans = fill_missing_with_nan(line.rstrip('\r\n'))
+                line_filled_w_nans = re.split(r'\t+',line_filled_w_nans)
+                data.append(line_filled_w_nans)
+            else:
+                continue
+
+    # Convert list into array
+    data_array = np.asarray(data)
+
+    # array into dataframe
+    dataf = pd.DataFrame(data_array, columns=input_rt_parameter.keys())
+    # check parameters
+    dataf = check_parameters(dataf, platform_code_short, input_rt_parameter, rt_file)
+
+    return dataf, platform_code
+
+def fill_missing_with_nan(line):
+    """
+    Solve issue with timesteps containing missing fields (case in Aurora files)
+    Loop through occurences of matched pattern (here consisting of 2 consecutive tabs)
+    and insert NaN between consecutive tabs. Since re.search looks for first location of pattern)
+    insert, function iterate until expression does not produce a match
+    """
+    while re.search(r'\t\t', line) is not None:
+        m=re.search(r'\t\t', line)
+        re.sub(r'\t\t',line[:m.start()+1]+'NaN'+line[m.end()-1:], line)
+        line=line[:m.start()+1]+'NaN'+line[m.end()-1:]
+
+    return line
+
+def get_time(dataf):
+    """
+    Convert date and time data object into datetime
+    Input : data Frame
+    Return:
+        - time:array of decimal time from 1950-01-01T00:00:00
+        - dtime: array of datetime object
+    """
+    epoch = datetime(1950, 1, 1)
+    time = []
+    time_long = data['PcDate'] + ' ' + data['PcTime']
```

Comment:
dataf or data ? line 299 the reference is data

Reply:
Well spotted. Actually the whole function is not use and should have been deleted.